### PR TITLE
fix: set max input length on search

### PR DESCRIPTION
**Pull request by MounirDhahri**
### Description

This PR sets the max input size on the search input component to be 100 - this is also the limit that we have from the backend. I believe a more solid approach would be to get the limit from the backend but this would turn this little effort into a significantly bigger one and I don't believe we need that, 100 chars is already BIG!

## Before

https://github.com/artsy/eigen/assets/11945712/d183a7f9-4588-4bc2-87fa-badb2f1dd430

## After

https://github.com/artsy/eigen/assets/11945712/3dc85e84-dc16-4083-999a-d2bd754bc10e

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- set max input length on search input field - mounir

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Review by gkartalis — APPROVED** [review]

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (set max input length on search input field - mounir)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 93e23c3a2c6daae5ab90629a100fd1c18dfb4cb0
</p>

**Comment by MounirDhahri** [comment]
Adding an additional comment to make sure we don't accidentally revert this

File: src/app/Components/SearchInput.tsx [hunk 1]
```diff
@@ -51,6 +51,9 @@ export const SearchInput = forwardRef<InputRef, SearchInputProps>(
               onClear?.()
               inputRef?.current?.focus()
             }}
+            // We only support up to 100 chars search in our backend,
+            // anything above that would lead to an error
```
Comment (gkartalis) [inline src/app/Components/SearchInput.tsx:55]: 👏

**Review by gkartalis — APPROVED** [review]

**Comment by MounirDhahri** [comment]
I will merge this and trigger new betas

**Pull request merged**